### ISSUE TITLE
Importing from node builtins gets messed up when libdef does so as well

### ISSUE THREAD

**Issue by iamdustan** [facebook/flow#3946]
Repro repo: https://github.com/iamdustan/flow-import-server-from-http-issue-when-express-exists

In short, if `flow-typed/npm/express_v4.x.x.x.js` is present then you can no longer do `import type { Server } from 'http';` in other app code.

To repro:

* `git clone https://github.com/iamdustan/flow-import-server-from-http-issue-when-express-exists.git`
* `yarn`
* `flow`

The only app code present in that repro is:

```jsx
/* @flow */
import type { Server } from 'http';
```

If you delete the `flow-typed` directory then Flow understands this to be the `Server` class from node's http module. Otherwise you'll get the following error:

```
  3: import type { Server } from 'http';
                   ^^^^^^ Named import from module `http`. This module has no named export called `Server`.
```

**Comment by larkintuckerllc** [comment]
Seems like this issue is duplicated in a newer issue: https://github.com/facebook/flow/issues/5113

Since, someone labeled this as a bug (and the title is more to the point), I will assume that this issue will the one that is going to be worked on.

That being said, it was nagging me about where the issue was (wasn't sure if it was a misunderstanding on my part).  To that end, I did some of my own testing to see if I could narrow down the problem.  Here is what I found:

1) The problem only appears to happen when you are using Library Definitions that imports a built in type, e.g., Server (from http) or URL (from url).  I verified this by writing two node modules (second pulled in a class definition from the first).  Then in my application, I could also pull in a class definition from the first module without the error).   In the same files, I would try to add in URL from the built-in url module and I would get the error (Library Definition file can import URL but the application cannot).

2) I was able to also confirm that the problem seems to only involve the use of Library Definition files as I used the technique of supplying an extra `.flow` file to provide the type definitions for my modules and I was able to import the build-in URL in both the module and application.

**Comment by gnprice** [comment]
This issue can be closed, because it was fixed in v0.143.0.

The cause of the symptoms above is that that libdef began with [this line](https://github.com/flow-typed/flow-typed/blob/e6aa8b0b44b99133d41fa5d391b1e482b8a732e4/definitions/npm/express_v4.x.x/flow_v0.32.x-/express_v4.x.x.js#L1) (linking to what was the latest version when this was filed):
```js
import type { Server } from 'http';
```

That's an import at the top level of a libdef file, outside any `declare module` block. This has always been invalid… and until v0.143.0 Flow's behavior was to give no error where it happens, and instead to give either very confusing errors (like above) elsewhere in your codebase, or else silently no errors at all where related types are used.

Since v0.143.0 Flow gives a nice clear error when a libdef does this, and when you move the import inside the `declare module` block all is well. See #6414 for details.
